partner_email_check/tests/test_partner_email_check.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2019 Komit <https://komit-consulting.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from unittest.mock import patch
+
+from odoo.exceptions import UserError, ValidationError
+from odoo.tests.common import SavepointCase
+from odoo.tools.misc import mute_logger
+
+
+class TestPartnerEmailCheck(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        # Checks are disabled during tests unless this key is set
```

Comment:
I don't get why we need this ctx key :thinking: 
Checks should be disabled by the flags set on the company.
If you want to keep them ON by default you can set them OFF in tests for the rest of the work by add demo data :wink:

Reply:
This is actually the same comment as https://github.com/OCA/partner-contact/pull/989#discussion_r665111764 since this key is only used there.

I think I get it now...

There are actually 3 checks performed:
-  Check email unique: only if set on company;
-  Check deliverability: only if set on company, performed with https://pypi.org/project/email-validator/ thus can't run without testing emails syntax as well;
- Check syntax: currently always performed unless in test mode without this key (test_partner_email_check), performed with https://pypi.org/project/email-validator/

So I guess the idea is not to break other modules' tests since not having this context key set (and being in test mode) is the only way not to check emails syntax.

Maybe the solution would be to add a third flag for the syntax check.
It should be off by default otherwise other modules' test may fail. Thus users must be aware that behavior changes with v13 since beforehand this check used to always be performed.

What do you think ?

Reply:
nope, the solution is simpler and I reported it above :wink: 

Just do this:

1. add demo data file
2. in that file call write function on all companies
3. set the flags to false
4. in your test class enable the flags

This way for the rest of the test suites (referred as "rest of the world" above) won't be impacted.

Reply:
of course, the alternative is that you make the flag false by default and you state in the README that it should be enabled